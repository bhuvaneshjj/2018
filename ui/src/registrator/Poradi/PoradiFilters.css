.PoradiFilters__kategorie {
  margin-left: 2em;
}
