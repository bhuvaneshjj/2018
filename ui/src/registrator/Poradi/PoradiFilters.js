import React from 'react';
import PropTypes from 'prop-types';
import { Button, ButtonGroup } from 'react-bootstrap';
import PopisekKategorie from '../../shared/Popisek/PopisekKategorie';
import KategorieFilter from '../Filterable/KategorieFilter';
import TextFilter from '../Filterable/TextFilter';
import Zobrazeno from '../Filterable/Zobrazeno';
import './PoradiFilters.css';

const PoradiFilters = ({
  kategorieFilters,
  kategorieSubFilters,
  numberOfItems,
  textFilter,
  onTextFilterChange
}) => (
  <div>
    <TextFilter filter={textFilter} onChange={onTextFilterChange} />

    <ButtonGroup className="PoradiFilters__kategorie">
      {kategorieFilters.map(({ active, typ, onClick }) => (
        <KategorieFilter active={active} key={typ} typKategorie={typ} onClick={onClick} />
      ))}
    </ButtonGroup>

    <ButtonGroup className="PoradiFilters__sub-kategorie">
      {kategorieSubFilters.map(({ active, kategorie, onClick }) => (
        <span
          className="PoradiFilters__sub-kategorie__item Bootstrap-buttons--active"
          key={kategorie.id}
        >
          <Button
            active={active}
            className={`PoradiFilters--${kategorie.typ}--${active ? 'active' : 'inactive'}`}
            onClick={onClick}
          >
            <PopisekKategorie {...kategorie} />
          </Button>
        </span>
      ))}
    </ButtonGroup>

    <Zobrazeno numberOfItems={numberOfItems} />
  </div>
);

PoradiFilters.propTypes = {
  kategorieFilters: PropTypes.arrayOf(
    PropTypes.shape({
      active: PropTypes.bool,
      typ: PropTypes.string.isRequired,
      onClick: PropTypes.func.isRequired
    }).isRequired
  ).isRequired,
  kategorieSubFilters: PropTypes.arrayOf(
    PropTypes.shape({
      active: PropTypes.bool,
      id: PropTypes.string.isRequired,
      typ: PropTypes.string.isRequired,
      onClick: PropTypes.func.isRequired
    }).isRequired
  ).isRequired,
  kategorieSubFiltersVisible: PropTypes.bool.isRequired,
  numberOfItems: PropTypes.number.isRequired,
  textFilter: PropTypes.string,
  onTextFilterChange: PropTypes.func.isRequired
};

export default PoradiFilters;
