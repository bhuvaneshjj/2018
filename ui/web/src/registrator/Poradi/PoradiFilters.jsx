import React from 'react';
import PropTypes from 'prop-types';
import { ButtonGroup } from 'react-bootstrap';
import KategorieFilter from '../Filterable/KategorieFilter';
import TextFilter from '../Filterable/TextFilter';
import Zobrazeno from '../Filterable/Zobrazeno';
import KategorieSubFilter from './KategorieSubFilter';
import './PoradiFilters.css';

const PoradiSubFilters = ({ kategorieSubFilters }) => {
  const { length } = kategorieSubFilters;
  if (length >= 6) {
    const half = length / 2;
    return (
      <div className="PoradiFilters__sub-kategorie">
        <div>
          {kategorieSubFilters.slice(0, half).map(({ id, ...props }) => (
            <KategorieSubFilter key={id} {...props} />
          ))}
        </div>
        <div>
          {kategorieSubFilters.slice(half, length).map(({ id, ...props }) => (
            <KategorieSubFilter key={id} {...props} />
          ))}
        </div>
      </div>
    );
  }
  return (
    <div className="PoradiFilters__sub-kategorie">
      {kategorieSubFilters.map(({ id, ...props }) => (
        <KategorieSubFilter key={id} {...props} />
      ))}
    </div>
  );
};

PoradiSubFilters.propTypes = {
  kategorieSubFilters: PropTypes.arrayOf(
    PropTypes.shape({
      active: PropTypes.bool,
      id: PropTypes.string.isRequired,
      typ: PropTypes.string.isRequired,
      onClick: PropTypes.func.isRequired
    }).isRequired
  ).isRequired
};

const PoradiFilters = ({
  kategorieFilters,
  kategorieSubFilters,
  kategorieSubFiltersVisible,
  numberOfItems,
  textFilter,
  onTextFilterChange
}) => (
  <div className="PoradiFilters">
    <TextFilter filter={textFilter} onChange={onTextFilterChange} />

    <ButtonGroup className="PoradiFilters__kategorie">
      {kategorieFilters.map(({ active, typ, onClick }) => (
        <KategorieFilter active={active} key={typ} typKategorie={typ} onClick={onClick} />
      ))}
    </ButtonGroup>

    {kategorieSubFiltersVisible && <PoradiSubFilters kategorieSubFilters={kategorieSubFilters} />}

    <Zobrazeno numberOfItems={numberOfItems} />
  </div>
);

PoradiFilters.propTypes = {
  kategorieFilters: PropTypes.arrayOf(
    PropTypes.shape({
      active: PropTypes.bool,
      typ: PropTypes.string.isRequired,
      onClick: PropTypes.func.isRequired
    }).isRequired
  ).isRequired,
  kategorieSubFilters: PropTypes.arrayOf(
    PropTypes.shape({
      active: PropTypes.bool,
      id: PropTypes.string.isRequired,
      typ: PropTypes.string.isRequired,
      zkratka: PropTypes.string.isRequired,
      onClick: PropTypes.func.isRequired
    }).isRequired
  ).isRequired,
  kategorieSubFiltersVisible: PropTypes.bool.isRequired,
  numberOfItems: PropTypes.number.isRequired,
  textFilter: PropTypes.string,
  onTextFilterChange: PropTypes.func.isRequired
};

PoradiFilters.defaultProps = {
  textFilter: undefined
};

export default PoradiFilters;
