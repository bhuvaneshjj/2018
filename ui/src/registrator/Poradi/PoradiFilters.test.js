import React from 'react';
import renderer from 'react-test-renderer';
import PoradiFilters from './PoradiFilters';

const kategorieFilters = [
  { active: true, typ: 'maraton', onClick: jest.fn() },
  { typ: 'cyklo', onClick: jest.fn() }
];
const kategorieSubFilters = [
  { id: '1', active: true, typ: 'maraton', onClick: jest.fn() },
  { id: '2', typ: 'maraton', onClick: jest.fn() },
  { id: '3', typ: 'maraton', onClick: jest.fn() }
];

it('renders kategorie filters', () => {
  const component = renderer.create(
    <PoradiFilters
      kategorieFilters={kategorieFilters}
      kategorieSubFilters={[]}
      kategorieSubFiltersVisible={false}
      numberOfItems={10}
      textFilter=""
      onTextFilterChange={jest.fn()}
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
});

it('renders kategorie subFilters', () => {
  const component = renderer.create(
    <PoradiFilters
      kategorieFilters={[kategorieFilters[0]]}
      kategorieSubFilters={kategorieSubFilters}
      kategorieSubFiltersVisible={true}
      numberOfItems={2}
      textFilter=""
      onTextFilterChange={jest.fn()}
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
});
