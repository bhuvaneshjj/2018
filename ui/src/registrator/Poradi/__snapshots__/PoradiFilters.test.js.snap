// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders kategorie filters 1`] = `
<div
  className="PoradiFilters"
>
  <input
    className="TextFilter__input"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Filtr na příjmení a jméno"
    type="text"
    value=""
  />
  <div
    className="PoradiFilters__kategorie btn-group"
  >
    <span
      className="KategorieFilter Bootstrap-buttons--active"
    >
      <button
        className="KategorieFilter--maraton--active btn btn-default active"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          style={
            Object {
              "fontSize": "100%",
            }
          }
        >
          <img
            alt="maraton"
            height={30}
            src="maraton.png"
            title="maraton"
          />
           
          maraton
        </span>
      </button>
    </span>
    <span
      className="KategorieFilter Bootstrap-buttons--active"
    >
      <button
        className="KategorieFilter--cyklo--inactive btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          style={
            Object {
              "fontSize": "100%",
            }
          }
        >
          <img
            alt="cyklo"
            height={30}
            src="cyklo.png"
            title="cyklo"
          />
           
          cyklo
        </span>
      </button>
    </span>
  </div>
  <span
    className="Zobrazeno"
  >
    zobrazeno: 
    <span
      className="badge"
    >
      10
    </span>
  </span>
</div>
`;

exports[`renders kategorie subFilters 1`] = `
<div
  className="PoradiFilters"
>
  <input
    className="TextFilter__input"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Filtr na příjmení a jméno"
    type="text"
    value=""
  />
  <div
    className="PoradiFilters__kategorie btn-group"
  >
    <span
      className="KategorieFilter Bootstrap-buttons--active"
    >
      <button
        className="KategorieFilter--maraton--active btn btn-default active"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          style={
            Object {
              "fontSize": "100%",
            }
          }
        >
          <img
            alt="maraton"
            height={30}
            src="maraton.png"
            title="maraton"
          />
           
          maraton
        </span>
      </button>
    </span>
  </div>
  <div
    className="PoradiFilters__sub-kategorie"
  >
    <span
      className="KategorieSubFilter Bootstrap-buttons--active"
    >
      <button
        className="KategorieSubFilter--maraton--active btn btn-default active"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          style={
            Object {
              "fontSize": "85%",
            }
          }
        >
           (1M)
        </span>
      </button>
    </span>
    <span
      className="KategorieSubFilter Bootstrap-buttons--active"
    >
      <button
        className="KategorieSubFilter--maraton--inactive btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          style={
            Object {
              "fontSize": "85%",
            }
          }
        >
           (2M)
        </span>
      </button>
    </span>
    <span
      className="KategorieSubFilter Bootstrap-buttons--active"
    >
      <button
        className="KategorieSubFilter--maraton--inactive btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          style={
            Object {
              "fontSize": "85%",
            }
          }
        >
           (3M)
        </span>
      </button>
    </span>
  </div>
  <span
    className="Zobrazeno"
  >
    zobrazeno: 
    <span
      className="badge"
    >
      2
    </span>
  </span>
</div>
`;
